command/get_conversation_replies.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright (C) 2018 Expedia Group.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/ExpediaGroup/flyte-slack/client"
+	"github.com/HotelsDotCom/flyte-client/flyte"
+	"github.com/slack-go/slack"
+	"strings"
+)
+
+var (
+	getConversationRepliesEventDef       = flyte.EventDef{Name: "GetConversationRepliesSuccess"}
+	getConversationRepliesFailedEventDef = flyte.EventDef{Name: "GetConversationRepliesFailed"}
+)
+
+type GetConversationRepliesInput struct {
+	ThreadTimestamp string `json:"threadTimestamp"`
+	ChannelId       string `json:"channelId"`
+}
+
+type GetConversationRepliesOutput struct {
+	Message []slack.Message `json:"messages"`
+}
+
+type GetConversationRepliesErrorOutput struct {
+	Error string `json:"error"`
+}
+
+func GetConversationReplies(slack client.Slack) flyte.Command {
+	return flyte.Command{
+		Name:         "GetConversationReplies",
+		OutputEvents: []flyte.EventDef{getConversationRepliesEventDef, getConversationRepliesFailedEventDef},
+		Handler:      getConversationRepliesHandler(slack),
+	}
+}
+
+func getConversationRepliesHandler(slack client.Slack) func(json.RawMessage) flyte.Event {
+	return func(rawInput json.RawMessage) flyte.Event {
+
+		input := GetConversationRepliesInput{}
+		if err := json.Unmarshal(rawInput, &input); err != nil {
+			return flyte.NewFatalEvent(fmt.Sprintf("invalid input: %v", err))
+		}
+
+		errorMessages := []string{}
+		if input.ThreadTimestamp == "" {
+			errorMessages = append(errorMessages, "missing threadTimestamp field")
+		}
+		if input.ChannelId == "" {
+			errorMessages = append(errorMessages, "missing channel id field")
+		}
+		if len(errorMessages) != 0 {
+			return newGetConversationRepliesFailedEvent(strings.Join(errorMessages, ", "))
+		}
```

Comment:
I would do this a bit differently.

I would first extract all of this in a separate method, e.g `validateInput`. Idiomatically, functions like these in Go are better suited to return an error. In an OOP language you'd use something like `isValid` and return a boolean, this might feel weird this way but it will read much better in code.

For example:

```
return func(rawInput ..) flyte.Event {
  in := conversationInput{}
  if err := json.Unmarshal(..) {
    // err
  }

  if err = validateInput(in); err != nil {
     return newFailedEvent(err);
  }

  ...

func validateInput(..) error {
   if in.ThreadStamp == "" { 
     return errors.New("missing threadTimestamp");
  }

 ....
```

While you won't be able to offer everything at once, it will read better in the codebase.